ignite/pkg/cosmosgen/templates/module/module.ts.tpl

```diff
@@ -79,12 +79,23 @@ class SDKModule {
 	public query: ReturnType<typeof queryClient>;
 	public tx: ReturnType<typeof txClient>;
 	
-	public registry: Array<[string, GeneratedType]>;
+	public registry: Array<[string, GeneratedType]> = [];
 
 	constructor(client: IgniteClient) {		
 	
-		this.query = queryClient({ addr: client.env.apiURL });
-		this.tx = txClient({ signer: client.signer, addr: client.env.rpcURL, prefix: client.env.prefix ?? "cosmos" });
+		this.query = queryClient({ addr: client.env.apiURL });		
+		const methods = txClient({ signer: client.signer, addr: client.env.rpcURL, prefix: client.env.prefix ?? "cosmos" });
+		this.tx=methods;
+		for (const method in methods) {
+			this.tx[method]=methods[method].bind(this.tx)
+		}
+		client.on('signer-changed',(signer) => {			
+			const methods = txClient({ signer: client.signer, addr: client.env.rpcURL, prefix: client.env.prefix ?? "cosmos" });
+			this.tx=methods;
+			for (const method in methods) {
+				this.tx[method]=methods[method].bind(this.tx)
+			}
+		})
 	}
```

Comment:
Maybe the TX update logic could be moved to a separate method? To something in the lines of:

```js
constructor(client: IgniteClient) {
    this.query = queryClient({ addr: client.env.apiURL })
    this.updateTX(client)
    client.on('signer-changed', (signer) => { this.updateTX(client) })
}

private updateTX(client: IgniteClient) {
    const methods = txClient({
        signer: client.signer,
        addr: client.env.rpcURL,
        prefix: client.env.prefix ?? "cosmos",
    })
	
    this.tx = methods
	for (let m in methods) {
		this.tx[m] = methods[m].bind(this.tx)
	}
}
```

Reply:
let me check if type inference  still works correctly after that